boards/nxp/mimxrt700_evk/board.c

```diff
@@ -0,0 +1,263 @@
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm17), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM17);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm17Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm18), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM18);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm18Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm19), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM19);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm19Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm20), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM20);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm20Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio0);
+	RESET_ClearPeripheralReset(kGPIO0_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio1);
+	RESET_ClearPeripheralReset(kGPIO1_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio2), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio2);
+	RESET_ClearPeripheralReset(kGPIO2_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio3), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio3);
+	RESET_ClearPeripheralReset(kGPIO3_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio4), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio4);
+	RESET_ClearPeripheralReset(kGPIO4_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio5), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio5);
+	RESET_ClearPeripheralReset(kGPIO5_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio6), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio6);
+	RESET_ClearPeripheralReset(kGPIO6_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio7), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio7);
+	RESET_ClearPeripheralReset(kGPIO7_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio8), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio8);
+	RESET_ClearPeripheralReset(kGPIO8_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio9), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio9);
+	RESET_ClearPeripheralReset(kGPIO9_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio10), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio10);
+	RESET_ClearPeripheralReset(kGPIO10_RST_SHIFT_RSTn);
+#endif
+
+	return 0;
+}
+
+static void GlikeyWriteEnable(GLIKEY_Type *base, uint8_t idx)
+{
+	(void)GLIKEY_SyncReset(base);
+
+	(void)GLIKEY_StartEnable(base, idx);
+	(void)GLIKEY_ContinueEnable(base, GLIKEY_CODEWORD_STEP1);
+	(void)GLIKEY_ContinueEnable(base, GLIKEY_CODEWORD_STEP2);
+	(void)GLIKEY_ContinueEnable(base, GLIKEY_CODEWORD_STEP3);
+	(void)GLIKEY_ContinueEnable(base, GLIKEY_CODEWORD_STEP_EN);
+}
+
+static void GlikeyClearConfig(GLIKEY_Type *base)
+{
+	(void)GLIKEY_SyncReset(base);
+}
+
+/* Disable the secure check for AHBSC and enable periperhals/sram access for masters */
+static void BOARD_InitAHBSC(void)
+{
+#if defined(CONFIG_SOC_MIMXRT798S_CPU0)
```

Comment:
can probably use if() here as well

Reply:
if this is changed, the compiler will report an error, CONFIG_SOC_MIMXRT798S_CPU0 cannot be found